src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/FixLegacyResourceDesignerStep.cs

```diff
@@ -105,10 +107,10 @@ internal override bool ProcessAssemblyDesigner (AssemblyDefinition assembly)
 			return true;
 		}
 
-		Dictionary<string, MethodDefinition> BuildResourceDesignerPropertyLookup (TypeDefinition type)
+		Dictionary<string, MethodDefinition> BuildResourceDesignerPropertyLookup (TypeDefinition type, StringComparer comparer = null)
```

Comment:
I wonder if this method would be more efficient if it *filled* two `Dictionary<string, MethodDefinition>` instances, so that it could iterate over the `TypeDefinition` once?

```csharp
void BuildResourceDesignerPropertyLookup (TypeDefinition type, out Dictionary<string, MethodDefinition> lookup, out Dictionary<string, MethodDefinition> lookupCaseInsensitive) => …
```

Maybe it wouldn't be more efficient, but it *could* allow more useful warnings if/when we encounter duplicate keys in the lookupCaseInsensitive map.

Though maybe that isn't the right place to warn either, if e.g. the value is never actually used.  The better place to warn might be in `FixBody()`…?

Reply:
I was planning to refactor this to populate both in the one call. It would make more sense than iterating over the list twice.